lib/AST/RequirementMachine/Rule.cpp

```diff
@@ -43,6 +43,33 @@ Optional<Symbol> Rule::isPropertyRule() const {
   return property;
 }
 
+Optional<Requirement> Rule::getPropertyRequirement(Type subject) const {
```

Comment:
This looks a lot like the isPropertyRule() part of the getRequirementFromRule lambda in RequirementBuilder::addRequirement()… I wonder if we can merge them together somehow. Or maybe this can just be a static method in this file, since it’s only used in one place.